mono/eglib/gmodule-win32.c

```diff
@@ -140,6 +140,45 @@ g_module_symbol (GModule *module, const gchar *symbol_name, gpointer *symbol)
 	}
 }
 
+gboolean
+g_module_address (void *addr, char **file_name, void **file_base, char **sym_name, void **sym_addr)
+{
+	HMODULE module;
+	/*
+	 * We have to cast the address because usually this func works with strings,
+	 * this being an exception.
+	 */
+	BOOL ret = GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS, (LPCWSTR)addr, &module);
```

Comment:
I believe GetModuleHandleEx increments the refcount of the module, so unless we then decrement the refcount at the end of this function we're leaking a ref. Not sure how much that matters in practice, but we should call FreeLibrary on the handle after we attempt to get the filename.

Reply:
It does. I added that in XP.
But FreeLibrary is also over-rated. A lot of .dlls don't behave well when unloaded anyway.